editor/src/clj/editor/resource.clj

```diff
@@ -207,13 +207,23 @@
 (defn make-memory-resource [workspace resource-type data]
   (MemoryResource. workspace (:ext resource-type) data))
 
-(defrecord ZipResource [workspace ^URI zip-uri name path data children]
+(defn- make-zip-resource-input-stream
+  ^InputStream [zip-resource]
+  (let [zip-file (ZipFile. ^File (io/as-file (:zip-uri zip-resource)))
+        zip-entry (.getEntry zip-file (:zip-entry zip-resource))]
+    (proxy [FilterInputStream] [(.getInputStream zip-file zip-entry)]
+      (close []
+        (let [^FilterInputStream this this]
+          (proxy-super close)
+          (.close zip-file))))))
+
+(defrecord ZipResource [workspace ^URI zip-uri name path zip-entry children]
```

Comment:
This is an unrelated fix to an issue discovered by Mathias a while back. We were loading the contents of any rependent `.zip` files into memory as byte-arrays. With native-extensions, these could potentially be quite large. With this change, we now open streams directly to the `.zip` files on disk when we need to read their contents. It does not appear to make a difference in project load times, but saves a bit of memory internally. Users's won't really see the memory savings, though, since the JVM will gladly make use of the extra heap space to store garbage. 😄